### ISSUE TITLE
Dispose Player and FlyLeaf on player.PlaybackCompleted error cross thread

### ISSUE THREAD

**Issue by sornanun** [SuRGeoNix/Flyleaf#71]
I need to dispose Player and FlyLeaf when video played complete.

But when I use this code to dispose error had occurred

```
// Dispose Player
if (_player != null)
{
    _player.Stop();
    _player.PlaybackCompleted -= _player_PlaybackCompleted;
    _player.OpenCompleted -= _player_OpenCompleted;
    _player.Dispose();
    _player = null;
}

// Dispose FlyLeaf
_flyleaf?.Dispose();  // <-- this line make error
_flyleaf = null;
```

Error message
`System.InvalidOperationException: Cross-thread operation not valid: Control '' accessed from a thread other than the thread it was created on.`

Can you add event for ended video and can dispose from main thread ?

**Comment by SuRGeoNix** [comment]
Yes disposing is not the easiest way yet. Normally, when the flyleaf control unloads should dispose itself so you don't have to dispose it manually. However, if you want to dispose it manually, you might need to set Mastter.PreventAutoDispose = true.
You don't need to dispose _flyleaf as _player should do it from its own dispose.
Finally, you need to ensure that if you dispose manually that the _player.Dispose() will be called from the UI thread (Dispatcher/Invoker) as it will need to access Dependency properties and UI Controls.

Let me know if you need more help with this.

**Comment by sornanun** [comment]
Thanks for reply SuRGeoNix.
I'm so sorry I don't quite understand.

I used MethodInvoker to Dispose player and Clear UserControl but in not working as my expect.

Now I use Timer for Dispose from UI Thread. Timer start before _player.Open("filepath") and when timer ticked I call to my [dispose()](https://github.com/SuRGeoNix/Flyleaf/issues/71#issue-973285339) function.

Then now it can Dispose from UI Thread.

But I must get video duration to set Timer.Interval before start. I use this function to get current video duration in seconds.
I'm not sure it works 100%, Because some video ex : MKV file type return negative value

```
public double TotalDuration()
{
    double duration = (double)_player?.Session?.VideoInfo?.Duration;
    if (duration > 0)
    {
        return duration / 10000000;
    }

    return 0.0;
}
```

**Comment by SuRGeoNix** [comment]
So the dispose worked fine by calling it from UI? What do you want to do? Why do you use a timer? If you want to dipose at the end just check the property changed of _player.Status to Ended or Stopped?
